src/devices/wifi/nm-iwd-manager.c

```diff
@@ -307,24 +332,105 @@ interface_added (GDBusObjectManager *object_manager, GDBusObject *object,
                  GDBusInterface *interface, gpointer user_data)
 {
 	NMIwdManager *self = user_data;
+	NMIwdManagerPrivate *priv = NM_IWD_MANAGER_GET_PRIVATE (self);
+	GDBusProxy *proxy;
+
+	if (!priv->running)
+		return;
 
-	set_device_dbus_object (self, interface, object);
+	g_return_if_fail (G_IS_DBUS_PROXY (interface));
+
+	proxy = G_DBUS_PROXY (interface);
+
+	if (!strcmp (g_dbus_proxy_get_interface_name (proxy),
+	             NM_IWD_DEVICE_INTERFACE)) {
+		set_device_dbus_object (self, proxy, object);
+		return;
+	}
+
+	if (!strcmp (g_dbus_proxy_get_interface_name (proxy),
+	             NM_IWD_KNOWN_NETWORK_INTERFACE)) {
+		KnownNetworkId *id;
+		KnownNetworkData *data;
+		NMIwdNetworkSecurity security;
+		gs_unref_variant GVariant *value;
```

Comment:
variables with `__attribute__((cleanup()))` must be initialized at definition. In this case ` = NULL`.

Otherwise, there is a compiler warning `-Werror=maybe-uninitialized`, in particular in combination with `-fexceptions`.

Reply:
It depends on the compiler and compiler options. Only because your current compilation doesn't report the warning, doesn't mean it's fine. Also, only because you initialize the variable before returning (and you know that there are no C++ exceptions that cause to leave the scope prematurely), is not enough to make the compiler happy.

Reply:
What I mean is: yes, there is no problem in practice, but the compiler warning is still emitted. You cannot argue with that, all compiler warnings must be avoided (worked-around) even if they are false-positives.

Reply:
Well, I simplified this whole variable away from the code now.  I didn't consider exceptions because this is compiled as C code but I guess a warning could be justified in that case.